Radarbox to AirNav Radar refactor

ketilmo opened the pull request:
I have refactored the relevant code to reflect Radarbox's name change to AirNav Radar. I have also upgraded the service OS to Bookworm.

* Created a new service named `airnav-radar` to reflect the name change of the service
* Created a breaking change-warning in the logs of the previous service if `RADARBOX_KEY` is set. (If `RADARBOX_KEY` is not set, the legacy `radarbox` service shuts down automatically.
* Changed startup procedure of service to wait for the key to be set in thevariable before starting to avoid orphan IDs
* Updated docs
* Updated OS to bookworm

I have considered various options for migrating from the old to the new service name. Since many balena users specifies variables per service, it's not always possible to gracefully read these from a service with a new name. To make the change consistent and stable, I found the approach above to work best. If the old service `radarbox` finds the variable `RADARBOX_KEY` to be set, it will periodically print out a "Changes required" message in the log, including instructions. If this variable is not set, it will shut down the container.

When this transitional period has run for a while (for instance, a year), the old radarbox service can be gracefully removed from the repo.

Requesting your review, @shawaj and @Teko012

shawaj commented:
Looks good to me. The only thing I would say is that would it make sense to instead have it handle the situation where the device variables are set for every container? 

I.e. have airnav container allow `RADARBOX_KEY` if it finds it too.

Because then it will just work for people that update without seeing the breaking change that already have the key configured for all containers?

ketilmo commented:
Thanks so much for your quick review, @shawaj! That’s a brilliant idea. I will make the suggested changes and update the PR accordingly.

Teko012 commented:
Yes, I agree. I would get rid of the radarbox container fully, and just support it in the new one if we can.

ketilmo commented:
Thanks for the feedback, @Teko012! Good suggestion. I will modify the PR and come back. 😊

ketilmo commented:
I have changed the logic a bit, trying different approaches to find the legacy `RADARBOX_KEY` if available.

* If the `AIRNAV_RADAR_KEY` is found, the new service will start without further action.
* If  both the `AIRNAV_RADAR_KEY` _and_ the legacy `RADARBOX_KEY` are found, the service will print a reminder to delete the legacy key during startup.
* If `AIRNAV_RADAR_KEY` is _not_ found `RADARBOX_KEY` is set for all services, the `RADARBOX_KEY` will be exported to `AIRNAV_RADAR_KEY` so the startup can continue.
* The edge case is when `RADARBOX_KEY` is set as a service variable for the `radarbox` container only. There is no easy way to fetch the variable in this case automatically, as it is only available from within the `radarbox` container. I have implemented a workaround: Start a minimal service in the legacy `radarbox` container, serving the `RADARBOX_KEY` variable over HTTP so `airnav-radar` can fetch it.
* As long as the `RADARBOX_KEY` exists, the legacy `radarbox` service will serve a periodic migration message in the console. Once this variable is deleted, the legacy service will shutdown gracefully. Continue running the `radarbox` container in a transitional phase will also ensure that setups with modified docker-compose.yml files will get alerted about the breaking change.
* I have also robustified the `showkey.sh` service a bit.

Teko012 commented:
@shawaj Do you think we should handle the edge case, or should we just have it as a breaking change in case someone only set that variable for that container?

shawaj commented:
yeah i am not sure - i get the idea of not breaking stuff, but maybe overkill to keep an unneeded container. i think personally I would just remove the radarbox container altogether and just have the renamed one. and if someone has a `RADARBOX_KEY` scoped to only radarbox container, they would just have to change the key.

But I don't really have any idea/visibility of how many people that might effect. My gut says that most people will just leave it as the default in balenaCloud when setting up device variables (which is to have it scoped to all containers) and so it would likely be very few people. But I could be completely wrong

ketilmo commented:
I agree with you. My hunch is that some users will have the key set as a service variable, but hopefully, they will pick up on the service change by seeing the notice in the console. I have made the changes based on our discussion. Feel free to have a new look.

Teko012 commented:
LGTM :)

ketilmo commented:
Thanks @Teko012! Much appreciated :)
